### PR TITLE
Test improvements to increase coverage

### PR DESCRIPTION
**Pull request by cgkoutzigiannis**
e.g. Small test improvements to increase the coverage on errors.py and fpdf.py.

**Checklist**:

- [x] The GitHub pipeline is OK (green), 
      meaning that both `pylint` (static code analyzer) and `black` (code formatter) are happy with the changes of this PR.

- [x] A unit test is covering the code added / modified by this PR

- [x] This PR is ready to be merged 

- [ ] In case of a new feature, docstrings have been added, with also some documentation in the `docs/` folder

- [ ] A mention of the change is present in `CHANGELOG.md`

- [ ] The PR description or comment contains a picture of a cute animal (not mandatory but encouraged 😉)

I can see in the coverage report that there are a few places where we could add new tests or improve the existing ones to increase the coverage. Do you think I am going in the right direction here in terms of the changes?
Also I am not sure if it's worth changing the CHANGELOG.md file for this change.

### CODE REVIEW

File: test/errors/test_FPDF_errors.py [hunk 1]
```diff
@@ -21,9 +21,11 @@ def test_encoding_exception():
     pdf = fpdf.FPDF()
     pdf.add_page()
     pdf.set_font("Helvetica", size=15)
-    with pytest.raises(FPDFUnicodeEncodingException):
+    with pytest.raises(FPDFUnicodeEncodingException) as e:
         pdf.cell(txt="Joséō")
         # This should through an error since Helvetica is a latin-1 encoder and the ō is out of range.
+    msg = 'Character "ō" at index 4 in text is outside the range of characters supported by the font used: "helvetica". Please consider using a font that supports UTF-8 encoding.'
```
Comment (gmischler) [inline test/errors/test_FPDF_errors.py:27]: While you're at it right here, could you maybe also correct that error message?
It should simply say "Please consider using a Unicode font". Unlike traditional fonts, Unicode fonts are not specific to any encoding (such as "utf-8").

Comment (cgkoutzigiannis) [reply test/errors/test_FPDF_errors.py:27]: Should I change it on errors.py where the exception is defined or just change the test case?
https://github.com/PyFPDF/fpdf2/blob/3c29cbab17fdfec17efcc5c39ab99440291c6556/fpdf/errors.py#L47-L52

Comment (Lucas-C) [reply test/errors/test_FPDF_errors.py]: Yes, the change must be made in `errors.py` :)

File: test/errors/test_FPDF_errors.py [hunk 2]
```diff
@@ -21,9 +21,12 @@ def test_encoding_exception():
     pdf = fpdf.FPDF()
     pdf.add_page()
     pdf.set_font("Helvetica", size=15)
-    with pytest.raises(FPDFUnicodeEncodingException):
+    with pytest.raises(FPDFUnicodeEncodingException) as e:
```
Comment (Lucas-C) [inline test/errors/test_FPDF_errors.py:24]: Could you rename this variable into `error`, which is a lot more explicit, please?

File: test/metadata/test_display_mode.py [hunk 3]
```diff
@@ -30,6 +31,24 @@ def test_setting_all_zoom(zoom, tmp_path):
...
+        h=0,
+        border=1,
+        txt="hello world",
+        fill=False,
+        link="",
+        new_x="LMARGIN",
+        new_y="NEXT",
+    )
+    with pytest.raises(FPDFException):
+        doc.set_display_mode("foo")
```
Comment (Lucas-C) [inline test/metadata/test_display_mode.py:49]: 👍

**Comment by Lucas-C** [comment]
Thank you @cgkoutzigiannis !

@allcontributors please add @cgkoutzigiannis for code

**Pull request merged**